src/main/resources/Macros/KanbanAWMSource.xml

```diff
@@ -115,13 +119,17 @@
   #if($request.debug)
   /* {{{ $results }}} */
   #end
-  #set($categoryField = $xwiki.getDocument($className).getxWikiClass().get($category))
-  #set($categories = $categoryField.getListValues())
+  #if ($categoryValues)
+    #set ($categories = $categoryValues)
```

Comment:
Since this is free user input, what happens if I type the wrong values? As in `Todos` instead of `Todo`

Reply:
Wrong values are still displayed as a column/category, but they won't have any cards to begin with.

Dragging a card in the wrongly named category WILL modify the field in the object, so it can lead to unintended behavior by way of propagating a mistake/typo (or including a category value which contains ```,```, as commas aren't escaped when splitting).

I think it's impossible to differentiate between 'correct' and 'wrong' values for String fields, which don't have a 'correct' value set. The only check would be if there are objects which use that value, but it would break when categories can be empty (in cases when there are no 'ToDo' tasks for example, if the status is only a String).

Or maybe only enforce these rules if there is a value set, like a DBList?

Reply:
Could you maybe display a category value only if it's part of `$categoryField.getListValues()`?